ui/app/js/controllers/rules_controller.js

```diff
@@ -216,6 +236,37 @@ function($scope, $routeParams, $location, $timeout, Rules, Search, $modal, $rout
     return true;
   };
 
+  $scope.$on('$routeChangeSuccess', function () {
+    $timeout(function () {
+      populateRulesProductsChannels();
+      var filterString = '';
+      if (!$routeParams.product && !$routeParams.channel) {
+        filterString = $rootScope.pr_ch_filter || 'All rules';
+        if ($rootScope.pr_ch_filter) {
+          changeLocationWithFilterParams($rootScope.pr_ch_filter);
+        }
+      }
+      if ($routeParams.product && !$routeParams.channel && $scope.pr_ch_options.includes($routeParams.product)) {
+        filterString = $routeParams.product;
+      } else if ($routeParams.product && !$routeParams.channel && !$scope.pr_ch_options.includes($routeParams.product)) {
+        filterString = 'All rules';
+      }
+      if ($routeParams.product && $routeParams.channel && $scope.pr_ch_options.includes($routeParams.product + ',' + $routeParams.channel)) {
+        filterString = $routeParams.product + ',' + $routeParams.channel;
+      } else if ($routeParams.product && $routeParams.channel && !$scope.pr_ch_options.includes($routeParams.product + ',' + $routeParams.channel)) {
+        filterString = 'All rules';
+      }
+      $scope.pr_ch_filter = filterString;
+      $rootScope.pr_ch_filter = $scope.pr_ch_filter;
+    }, 2000);
```

Comment:
As I'm sure you know, this timeout of 2000 is why the location isn't updated immediately upon page load. Why do you need it, though? Why not just execute all of this code directly without the $timeout()? I tried that locally and everything appeared to work still.